syncapi/storage/postgres/syncserver.go

```diff
@@ -642,12 +646,32 @@ func (d *SyncServerDatasource) getResponseWithPDUsForCompleteSync(
 	return res, toPos, joinedRoomIDs, err
 }
 
+func removeIgnoredUserEvents(events []types.StreamEvent, ignoredUsers []string) []types.StreamEvent {
```

Comment:
I'd be more inclined to implement this filtering in the SQL request, so that we return the number of events dictated by the filter as often as possible. FYI you can do that with PostgreSQL by giving an array to `QueryContext` and add `AND sender != ALL($X)` to the corresponding SQL query (obviously replace `X` with the order of the argument).